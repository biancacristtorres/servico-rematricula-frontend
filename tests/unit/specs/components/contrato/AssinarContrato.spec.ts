import {
  AssinarContratoFactory,
} from '../../../factory/components/contrato/AssinarContratoFactory';

describe('Componente de Assinar Contrato', () => {
  const factory = new AssinarContratoFactory();

  beforeEach(() => {
    factory.prepararContainerDeInjecaoDaApi();
  });

  afterEach(() => {
    factory.restaurarContainerDeInjecaoDaApi();
  });

  it('Ao assinar contrato com sucesso, redireciona para Resumo', async (done) => {
    factory.dadoQueOContratoSejaAssinadoComSucesso();

    await factory.montarComponente();
    factory.componente.find('#assinar-contrato').trigger('click');

    factory.redirecionamento = (rota: string) => {
      expect(rota).toEqual('/Resumo');
      done();
    };
  });

  it('Ao assinar contrato com sucesso, uma action para limpar a store do aluno é chamada', async (done) => {
    factory.dadoQueOContratoSejaAssinadoComSucesso();

    await factory.montarComponente();
    factory.componente.find('#assinar-contrato').trigger('click');

    expect(factory.limparInformacoesDoAluno).toBeCalled();
    done();
  });

  it('Quando o contrato não pode ser assinado, redireciona para Sistema Indisponível', async (done) => {
    factory.dadoQueOContratoNaoPodeSerAssinado();

    await factory.montarComponente();
    factory.componente.find('#assinar-contrato').trigger('click');

    factory.redirecionamento = (rota: string) => {
      expect(rota).toEqual('/erro/SistemaIndisponivel');
      done();
    };
  });

  it('Quando não foi possível assinar o contrato, redireciona para Sistema Indisponível', async (done) => {
    factory.dadoQueAcontecaErroAoAssinarOContrato();

    await factory.montarComponente();
    factory.componente.find('#assinar-contrato').trigger('click');

    factory.redirecionamento = (rota: string) => {
      expect(rota).toEqual('/erro/SistemaIndisponivel');
      done();
    };
  });

  it('Ao assinar contrato, o estado de loading deve ser verdadeiro', async (done) => {
    factory.dadoQueOContratoSejaAssinadoComSucesso();

    await factory.montarComponente();

    expect(factory.vueInstance.carregando).toBeFalsy();

    factory.componente.find('#assinar-contrato').trigger('click');

    expect(factory.vueInstance.carregando).toBeTruthy();

    factory.redirecionamento = (rota: string) => {
      expect(rota).toEqual('/Resumo');
      done();
    };
  });

  it('Snapshot', async (done) => {
    expect(factory.componente).toMatchSnapshot();
    done();
  });
});
