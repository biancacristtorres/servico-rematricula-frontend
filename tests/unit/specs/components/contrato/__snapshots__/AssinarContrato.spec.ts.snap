// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de Assinar Contrato Snapshot 1`] = `
<vbtn-stub activeclass="v-btn--active" large="true" tag="button" type="button" id="assinar-contrato" color="primary" loading="true">
  Li, concordo e aceito os termos propostos
</vbtn-stub>
`;
